src/cpp/shared_core/FilePath.cpp

```diff
@@ -394,6 +394,22 @@ FilePath::FilePath(const std::wstring& absolutePath)
 }
 #endif
 
+FilePath::FilePath(const char* in_absolutePath) :
+   m_impl(in_absolutePath ? 
+            new Impl(fromString(in_absolutePath)) :
+            new Impl())
```

Comment:
This was effectively the solution I came up with the first time someone took a stab at this problem, but @kevinushey pointed out that `FilePath(accidentallyNull).completePath("blah")` would end up being a relative path from the cwd, rather than a child of the (relative or absolute) root path, and cause undesired behaviour.

Reply:
I had a vague memory that we didn't create an empty path for some reason but couldn't remember what it was. Thank you for refreshing my memory!

I think it makes more sense to handle that issue in `completePath`. How would you feel about either (a) propagating the emptiness of the path (i.e. a completion of an empty path will also be considered to be empty), with a warning logged, or (b) throwing an exception when trying to complete an empty path?

Reply:
I would be on board with logging a warning about it. We don't have hard evidence that this happens in the wild, and we're not set up for exceptions from our FilePath methods. I think we can examine a more holistic change to the codebase if we start seeing those warnings in diagnostics files.

Reply:
Great, I'll make that change.

Reply:
Update: it didn't go well, it turns out that we DO complete empty paths as a matter of course! Will need more investigation, which I'll do when I return.